app/src/main/java/com/sakaimobile/development/sakaiclient20/ui/fragments/AnnouncementsFragment.java

```diff
@@ -100,26 +104,49 @@ public void onCreate(@Nullable Bundle savedInstanceState) {
     public View onCreateView(LayoutInflater inflater, @Nullable ViewGroup container, @Nullable Bundle savedInstanceState) {
 
         View view = inflater.inflate(R.layout.fragment_announcements, null);
+
+        // setup recycler view
         announcementRecycler = view.findViewById(R.id.announcements_recycler);
         LinearLayoutManager layoutManager = new LinearLayoutManager(getContext());
         layoutManager.setOrientation(LinearLayoutManager.VERTICAL);
         announcementRecycler.setLayoutManager(layoutManager);
         announcementRecycler.setItemAnimator(new DefaultItemAnimator());
 
-        swipeRefreshLayout = view.findViewById(R.id.swiperefresh);
-        swipeRefreshLayout.setRefreshing(true);
+        // start the spinner
+        spinner = view.findViewById(R.id.progress_circular);
+        spinner.setVisibility(View.VISIBLE);
 
         createAdapter();
 
-        liveData.observe(getActivity(), announcements -> {
+        // update the recycler adapter when new announcements come
+        announcementLiveData.observe(getActivity(), announcements -> {
```

Comment:
Observations should always go in `onViewCreated` and be subscribed to using the view's lifecycle (using `getViewLifecycleOwner()`), instead of the parent context (here `getActivity()`). [This prevents double subscriptions by ensuring that the subscription is unregistered in `onDestroyView`](https://github.com/googlesamples/android-architecture-components/issues/47#issuecomment-395337503).